RELEASE.md

```diff
@@ -1,6 +1,14 @@
 # Appsody CLI Release Process
 Follow this process to create a new release of the Appsody CLI.
 
+# Dependencies
+The Appsody CLI depends on a number of assets:
+1. The `Appsody Controller`, which is available [here](https://github.com/appsody/controller). The controller version to pull is located in the `Makefile` as `CONTROLLER_VERSION` variable. The version is updated manually as needed, there is no automatic update when a new controller is release. However, the dependent controller version does need to be released first, as the build process for the Appsody CLI retrieves the controller binaries and does not build them. Once the dependent controller has been released, a Pull Request with the `CONTROLLER_VERSION` changes to the `Makefile` needs to be created and merged before continuing on with the CLI release.
+1. The `debian-builder` Docker image, which is built separately from [this repo](https://github.com/appsody/debian-builder). The `Makefile` has a variable that points to the latest version of this image, called `DOCKER_IMAGE_DEB`. This image is used during the deploy stage to generate the Debian installer for Appsody. 
+1. The `rpmbuilder` Docker image, which builds the RPM installer for Appsody. Currently, we use the image provided by `alectolytic`, and we point to it through the `Makefile` env var `DOCKER_IMAGE_RPM`.
+
+We expect the Docker images for the Debian and RPM installers to change very rarely. The Appsody controller may be released more frequently. Generally, a new release of the Controller requires a new release of the Appsody CLI.
+
 ### Create a GitHub release
```

Comment:
After moving the `# Dependencies` section up, the section headers seem to be a bit off. 
We could move the `# Appsody CLI Release Process` header down here, or create another top level header here called something like `# Step-by-step release guide` ?